pkg/git/libgit2/managed/ssh.go

```diff
@@ -191,21 +191,8 @@ func (t *sshSmartSubtransport) Action(transportOptionsURL string, action git2go.
 	}
 
 	sshConfig.HostKeyCallback = func(hostname string, remote net.Addr, key ssh.PublicKey) error {
-		marshaledKey := key.Marshal()
-		cert := &git2go.Certificate{
-			Kind: git2go.CertificateHostkey,
-			Hostkey: git2go.HostkeyCertificate{
-				Kind:         git2go.HostkeySHA256 | git2go.HostkeyRaw,
-				HashSHA256:   sha256.Sum256(marshaledKey),
-				Hostkey:      marshaledKey,
-				SSHPublicKey: key,
-			},
-		}
-
-		if len(opts.AuthOpts.KnownHosts) > 0 {
-			return KnownHostsCallback(hostname, opts.AuthOpts.KnownHosts)(cert, true, hostname)
-		}
-		return nil
+		keyHash := sha256.Sum256(key.Marshal())
+		return CheckKnownHost(hostname, opts.AuthOpts.KnownHosts, keyHash[:])
```

Comment:
Not sure if it can be a real issue, but with this change, we'll lose a check where the certificate is validated to be host key of kind SHA256. That's performed in `KnownHostsCallback()` only with this change.
The existing tests are for `KnownHostsCallback()` only which has this validation. Maybe we need separate tests for `CheckKnownHost()`?

Reply:
This was kind of intentional. As part of decommissioning unmanaged transport, only CheckKnownHost will be required, and therefore all the existing tests will need to be refactored.

I made sure the workflow was still valid and tests as-is, but avoided duplicating the tests - that soon would be removed anyway.